src/Microsoft.Health.Fhir.Api/Features/Routing/SearchPostReroutingMiddleware.cs

```diff
@@ -33,7 +33,7 @@ public async Task Invoke(HttpContext context)
                 && request.Method == "POST"
                 && request.Path.Value.EndsWith(KnownRoutes.Search, System.StringComparison.OrdinalIgnoreCase))
             {
-                if (request.ContentType is null || request.ContentType == "application/x-www-form-urlencoded")
+                if (request.ContentType is null || request.HasFormContentType)
```

Comment:
Possible to add a regression test for this bug?

Reply:
Added some UTs.